plugins/de.fraunhofer.ipa.rossystem.xtext/src/de/fraunhofer/ipa/rossystem/generator/LaunchFileCompiler_ROS2.xtend

```diff
@@ -78,6 +78,67 @@ def generate_launch_description():
 		return component.nameSpace.replaceFirst("/","");
 	}
 
+	def String is_comma(Integer count) {
+		return (count > 0) ? ",\n" : "\n"
+	}
+
+	def String compile_remappings_str(ComponentInterface component) {
+		var remap_str = "";
+		var count = component.rospublisher.length
+					+ component.rossubscriber.length
+					+ component.rosserviceserver.length
+					+ component.rosserviceclient.length
+					+ component.rosactionserver.length
+					+ component.rosactionclient.length;
```

Comment:
just realized this calculation does not make sense.

Reply:
is an open comma producing an error? if yes, you can create a boolean flag "HasRemaps" set to true if you enter one of the if conditions, then you can add the comma to all the remap_str modifications and at the end remove the last character.

I mean:


```
HasRemaps=false;

    for (rosPublisher : component.rospublisher) {
			if (!((prefix(NS)+rosPublisher.name).equals(compile_topic_name(rosPublisher.publisher, NS)))) {
				remap_str += "\t(\"" + rosPublisher.publisher.name + "\", \"" + rosPublisher.name + "\"),/n";
        HasRemaps=true;
			}
		}
		for (rosSubscriber : component.rossubscriber) {
			if (!((prefix(NS)+rosSubscriber.name).equals(compile_topic_name(rosSubscriber.subscriber, NS)))) {
				remap_str += "\t(\"" + rosSubscriber.subscriber.name + "\", \"" + rosSubscriber.name + "\"),/n";
				HasRemaps=true;
			}
		}

.....



if (HasRemaps){
  remap_str=remap_str.substring(0,remap_str.lenght-2)
  remap_str +="]"
}

```